torch/distributed/pipelining/_backward.py

```diff
@@ -320,6 +320,8 @@ def extract_tensors_with_grads(output_val, grad_val):
         torch.autograd.backward(
             stage_output_tensors, grad_tensors=output_grad_tensors  # type: ignore[arg-type]
         )
+        del stage_output_tensors
```

Comment:
after digging deeper into what the ref cycle was, I prefer to change this to `del extract_tensors_with_grads` which breaks the cycle and ends up freeing the tensors as well as other pyobjs that would have remained in a cycle if i only `del stage_output_tensors` and `del output_grad_tensors`

Reply:
i've updated the PR with a fix that is unfortunately a bit uglier, but arguably more correct.  If folks have a better idea, let me know

Reply:
ok- one more update: i don't really need to pass `stage_output_tensors` and `output_grad_tensors` into `extract_tensors_with_grads`.  I'll let those be captured in the cell.  I just need to ensure the cell will die, and for that i just have to break the cycle caused by the self-reference.  And that's a little less ugly.